src/superfine.ts

```diff
@@ -411,14 +442,21 @@ export var patch = function(lastNode, nextNode, container) {
   return nextNode
 }
 
-export var h = function(name, props) {
+/**
+ * create a new vdom node. vdom is a description of what DOM should look like using a tree of virtual nodes.
+ * @param name name of an element or function that returns vdom node
+ * @param props HTML props, SVG props, DOM events, lifecycle events and keys
+ * @param rest child nodes for an element
+ */
+export var h = function(
+  name: string | Function,
+  props?: any,
```

Comment:
@mindplay-dk existing definition for h are open to Props being a type. This works ok when declaring TS definitions separately. Any pointers on how I can achieve that using TS definitions in file?

Reply:
The only difference as far as I can figure, is the `h` function will have a body - other than that, you should be able to just copy and paste from the original `.d.ts`?

You'll want to exclude the `.d.ts` from the build (as you do already) and move to emitting a generated `.d.ts` instead - once you have all the type declarations moved over, the emitted `.d.ts` should be very close to the hand-written one we have now.

(I'm not sure if that's what you were asking?)